FreeRTOS-Plus/Source/Utilities/logging/logging_stack.h

```diff
@@ -37,25 +37,58 @@
 #include "logging_levels.h"
 
 /* Standard Include. */
+#include <stdio.h>
 #include <stdint.h>
 
+/* The macro definition for LIBRARY_LOG_NAME is for Doxygen
+ * documentation only. This macro is typically defined in only the
+ * <library>_config.h file or the demo_config.h file. */
+
+/**
+ * @brief The name of the library or demo to add as metadata in log messages
+ * from the library or demo.
+ *
+ * This metadata aids in identifying the module source of log messages.
+ * The metadata is logged in the format `[ <LIBRARY-NAME> ]` as a prefix to the
+ * log messages.
+ * Refer to #LOG_METADATA_FORMAT for the complete format of the metadata prefix in
+ * log messages.
+ */
+/* Check if LIBRARY_LOG_NAME macro has been defined. */
+#if !defined( LIBRARY_LOG_NAME )
+    #error "Please define LIBRARY_LOG_NAME for the library."
+#endif
+
 /* Metadata information to prepend to every log message. */
-#define LOG_METADATA_FORMAT    "[%s:%d] "
-#define LOG_METADATA_ARGS      __FUNCTION__, __LINE__
+#ifndef LOG_METADATA_FORMAT
+    #define LOG_METADATA_FORMAT    "[%s] [%s:%d] "                  /**< @brief Format of metadata prefix in log messages as `[<Logging-Level>] [<Library-Name>] [<Function-Name>:<Line-Number>]` */
+#endif
 
-/* Common macro for all logging interface macros. */
-#if !defined( DISABLE_LOGGING )
+#ifndef LOG_METADATA_ARGS
+    #define LOG_METADATA_ARGS    LIBRARY_LOG_NAME, __FUNCTION__, __LINE__ /**< @brief Arguments into the metadata logging prefix format. */
+#endif
 
 /* Prototype for the function used to print out.  In this case it prints to the
  * console before the network is connected then a UDP port after the network has
  * connected. */
-    extern void vLoggingPrintf( const char * pcFormatString,
-                                ... );
+extern void vLoggingPrintf( const char * pcFormatString,
```

Comment:
If this is a generic header file, intended for use with all FreeRTOS ports, then this line and the above comment shouldn't be here because they are specific to running FreeRTOS on Windows.

Reply:
Makes sense, removed the prototype from this generic header.